proto/v1/idp_service.proto

```diff
@@ -124,6 +131,34 @@ message UndeleteIdentityProviderRequest {
   string name = 1 [(google.api.field_behavior) = REQUIRED];
 }
 
+message TestIdentityProviderRequest {
+  // The identity provider to test connection including uncreated.
+  IdentityProvider identity_provider = 1;
+
+  oneof context {
+    OAuth2IdentityProviderTestRequestContext oauth2_context = 2;
+  }
+}
+
+message OAuth2IdentityProviderTestRequestContext {
+  // Authorize code from website.
+  string code = 1;
+}
+
+message TestIdentityProviderResponse {
+  string message = 1;
+  
+  oneof result {
+    OAuth2IdentityProviderTestResult oauth2_result = 2;
+  }
+}
+
+message OAuth2IdentityProviderTestResult {
+  string access_token = 1;
```

Comment:
Can we not return the token?

Reply:
The purpose of this API is to clearly show whether the configuration is correct and the main reason of error. 

As OAuth2's workflow is `code` -> `access_token` -> `userinfo`, I think it's better to return the result of each step.